naming/src/main/java/org/jboss/as/naming/NamingContext.java

```diff
@@ -242,15 +244,33 @@ public Object lookup(final Name name, boolean dereference) throws NamingExceptio
     }
 
     /** {@inheritDoc} */
-    public void bind(final Name name, Object object) throws NamingException {
+    public void bind(final Name name, final Object object) throws NamingException {
         check(name, Action.BIND);
 
         if(namingStore instanceof WritableNamingStore) {
             final Name absoluteName = getAbsoluteName(name);
+            final Object value;
             if (object instanceof Referenceable) {
-                object = ((Referenceable) object).getReference();
+                value = ((Referenceable) object).getReference();
+            } else {
+                value = object;
+            }
+            // The permissions check has already happened for the binding further permissions should be allowed
+            final NamingException e = AccessController.doPrivileged(new PrivilegedAction<NamingException>() {
```

Comment:
This needs a check to see if the security manager is active. AccessController.doPrivileged should only be used when it is absolutely required, as it is slow.